### ISSUE TITLE
Issues with an Ubuntu toolbox on pop_OS

### ISSUE THREAD

**Issue by spencergilbert** [lopukhov/unbox#5]
:wave:

`5.19.0-76051900-generic` and `Pop!_OS 22.04 LTS` 

The toolbox is created without issue...

```shell
~ 
❯ unbox create --engine podman --image docker.io/ubuntu:latest ubuntu

~ took 2s 
❯ unbox enter ubuntu
```

However, after entering the toolbox I'm unable to add arbitrary packages (as you can with `toolbx`)

```shell
spencer.gilbert in ~ 
❯ apt install rolldice
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package rolldice

spencer.gilbert in ~ 
❯ sudo apt install rolldice
bash: sudo: command not found

spencer.gilbert in ~ 
❯ apt update
E: setgroups 65534 failed - setgroups (1: Operation not permitted)
E: setegid 65534 failed - setegid (22: Invalid argument)
E: seteuid 100 failed - seteuid (22: Invalid argument)
E: setgroups 0 failed - setgroups (1: Operation not permitted)
Reading package lists... Done
W: chown to _apt:root of directory /var/lib/apt/lists/partial failed - SetupAPTPartialDirectory (22: Invalid argument)
W: chown to _apt:root of directory /var/lib/apt/lists/auxfiles failed - SetupAPTPartialDirectory (22: Invalid argument)
E: setgroups 65534 failed - setgroups (1: Operation not permitted)
E: setegid 65534 failed - setegid (22: Invalid argument)
E: seteuid 100 failed - seteuid (22: Invalid argument)
E: setgroups 0 failed - setgroups (1: Operation not permitted)
E: Method gave invalid 400 URI Failure message: Failed to setgroups - setgroups (1: Operation not permitted)
E: Method gave invalid 400 URI Failure message: Failed to setgroups - setgroups (1: Operation not permitted)
E: Method http has died unexpectedly!
E: Sub-process http returned an error code (112)

spencer.gilbert in ~ 
❯ apt upgrade
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
W: chown to _apt:root of directory /var/cache/apt/archives/partial failed - SetupAPTPartialDirectory (22: Invalid argument)
W: chown to _apt:root of directory /var/lib/apt/lists/auxfiles failed - SetupAPTPartialDirectory (22: Invalid argument)
```

Again, happy to provide more information and/or help PR a fix!

**Comment by lopukhov** [comment]
:wave:

I believe that I know why this happens, to try to avoid having to set capabilities the file `/proc/<pid>/setgroups` is set to `deny`. But this breaks some software and I did not think about it at the moment. 

Reading the [documentation](https://www.man7.org/linux/man-pages/man7/user_namespaces.7.html) I can see there are two options:

1. Set `CAP_SETUID` and `CAP_SETGID` on `unbox` and create the mapping without setting `setgroups` to `deny`
2. Spawn `newuidmap`/`newgidmap` and let it take care of it (as far as I know this is how `podman` does it).

I'd rather do the first one to avoid depending on external binaries as much as possible, but I am having trouble getting it to work right now. Tomorrow I will try again but if the second one does not incur in a significant performance penalty it could also be an option.

**Comment by lopukhov** [comment]
:wave: 

I have just released v0.3.0. This new version should fix this issue. 

The commit that fix the issue is e89d5fb83adf2c94e5240d7bd6bd3bc23658d9ac and it needed to spawn `newuidmap` and `newgidmap`. 

If the issue persists please let me know.
